### PR TITLE
fix: remove prometheus import from the top of the files

### PR DESCRIPTION
**Pull request by samsja**
We want to be able to remove prometheus from core dependency if needed. Therefore we need to remove the import from the top of the file

### CODE REVIEW

File: jina/serve/executors/__init__.py [hunk 1]
```diff
@@ -124,7 +122,12 @@ def _add_runtime_args(self, _runtime_args: Optional[Dict]):
             self.runtime_args = SimpleNamespace()
 
     def _init_monitoring(self):
-        if hasattr(self.runtime_args, 'metrics_registry'):
+        if (
+            hasattr(self.runtime_args, 'metrics_registry')
```
Comment (JoanFM) [inline jina/serve/executors/__init__.py:126]: can't we make sure `metrics_registry` is always in the `runtime_args`?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/executors/__init__.py (hunk 1 above)
Comment (samsja) [reply jina/serve/executors/__init__.py:126]: Yes we should you are right

Comment (JoanFM) [reply jina/serve/executors/__init__.py:126]: then let's do it here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/executors/__init__.py [hunk 2]
```diff
@@ -124,7 +122,12 @@ def _add_runtime_args(self, _runtime_args: Optional[Dict]):
             self.runtime_args = SimpleNamespace()
 
     def _init_monitoring(self):
-        if hasattr(self.runtime_args, 'metrics_registry'):
+        if not hasattr(self.runtime_args, 'metrics_registry'):
```
Comment (JoanFM) [inline jina/serve/executors/__init__.py:125]: this should not be done here. This should be handled in the setting of these `arguments`. `_init_monitoring` should not alter the `runtime_args`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/executors/__init__.py [hunk 3]
```diff
@@ -124,7 +122,12 @@ def _add_runtime_args(self, _runtime_args: Optional[Dict]):
...
+            self.runtime_args.metrics_registry = None
+
+        if self.runtime_args.metrics_registry:
+            from prometheus_client import Summary
+
```
Comment (JoanFM) [inline jina/serve/executors/__init__.py:130]: use ImportExtensions to guide the user in case not added

**Review by JoanFM — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -304,10 +304,33 @@ jobs:
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Prepare enviroment
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install wheel
+          pip install --no-cache-dir .
+      - name: Test basic import
+        run: python -c 'from jina import Executor,requests'
```
Comment (mapleeit) [inline .github/workflows/ci.yml:327]:
```suggestion
  import-test:
    runs-on: ubuntu-latest
    needs: [ commit-lint, lint-flake-8, code-injection ]
    strategy:
      fail-fast: false
      matrix:
        core: ['', 'true']
        perf: ['', 'true']
        exclude: 
          - core: 'true'
            perf: 'true'
    steps:
      - uses: actions/checkout@v2
      - name: Set up Python 3.7
        uses: actions/setup-python@v2
        with:
          python-version: 3.7
      - name: Prepare enviroment
        run: |
          python -m pip install --upgrade pip
          python -m pip install wheel
          pip install --no-cache-dir .
        env:
          JINA_PIP_INSTALL_CORE: ${{ matrix.core }}
          JINA_PIP_INSTALL_PERF: ${{ matrix.perf }}
      - name: Test basic import
        run: python -c 'from jina import Executor,requests'
```

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -157,6 +157,28 @@ jobs:
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Prepare enviroment
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install wheel
+          pip install --no-cache-dir .
+      - name: Test basic import
+        run: python -c 'from jina import Executor,requests'
+
```
Comment (mapleeit) [inline .github/workflows/cd.yml:181]:
```suggestion
  import-test:
    runs-on: ubuntu-latest
    needs: [ commit-lint, lint-flake-8, code-injection ]
    strategy:
      fail-fast: false
      matrix:
        core: ['', 'true']
        perf: ['', 'true']
        exclude: 
          - core: 'true'
            perf: 'true'
    steps:
      - uses: actions/checkout@v2
      - name: Set up Python 3.7
        uses: actions/setup-python@v2
        with:
          python-version: 3.7
      - name: Prepare enviroment
        run: |
          python -m pip install --upgrade pip
          python -m pip install wheel
          pip install --no-cache-dir .
        env:
          JINA_PIP_INSTALL_CORE: ${{ matrix.core }}
          JINA_PIP_INSTALL_PERF: ${{ matrix.perf }}
      - name: Test basic import
        run: python -c 'from jina import Executor,requests'
```

**Comment by mapleeit** [comment]
@samsja @JoanFM I optimized the `import-test` a little bit to cover the case of use `jina[core]` and `jina[perf]`, can you review if it makes sense?

**Comment by mapleeit** [comment]
btw, the two failed tests can be ignored. We need to release a patch to fix that.

**Review by mapleeit — APPROVED** [review]

**Pull request merged**